server/autotest_server/testers/ai/ai_tester.py

```diff
@@ -0,0 +1,116 @@
+import json
+import os
+import sys
+
+from ..tester import Test, Tester
+from ..specs import TestSpecs
+import subprocess
+from typing import Type
+from dotenv import load_dotenv
+
+
+class AiTest(Test):
+    def __init__(
+        self,
+        tester: "AiTester",
+        result: dict = None,
+    ):
+        self._test_name = result["title"]
+        self.message = result.get("message", "")
+        self.status = result["status"]
+        super().__init__(tester)
+
+    @property
+    def test_name(self) -> str:
+        """The name of this test"""
+        return self._test_name
+
+    @Test.run_decorator
+    def run(self) -> str:
+        """
+        Return a json string containing all test result information.
+        """
+        if self.status in ("success", "pass"):
+            return self.passed(message=self.message)
+        elif self.status == "partial":
+            return self.partially_passed(points_earned=1, message=self.message)
+        else:
+            return self.error(message=self.message)
+
+
+class AiTester(Tester):
+    def __init__(
+        self,
+        specs: TestSpecs,
+        test_class: Type[AiTest] = AiTest,
+        resource_settings: list[tuple[int, tuple[int, int]]] | None = None,
+    ) -> None:
+        """
+        Initialize a new AIFeedbackTester using the specifications in the specs
+        """
+        super().__init__(specs, test_class, resource_settings=resource_settings)
+        self.annotations = []
+        self.overall_comments = []
+
+    def call_ai_feedback(self) -> dict:
+        """
+        Call ai_feedback CLI using subprocess and arguments from self.specs.
+        Supports all standard arguments from ai_feedback.
+        """
+        results = {}
+        load_dotenv()
+        env = os.environ.copy()
+
+        test_data = self.specs["test_data"]
+        if isinstance(test_data, dict):
+            test_data = [test_data]
+        for test_group in test_data:
+            config = test_group.get("config", {})
+            title = test_group.get("title", "Unnamed Test")
```

Comment:
"title" isn't the best attribute name for this. Let's go with `test_label` (display text: "Test label") to better describe it. The default value can be "Generate AI Feedback"; you can add this to the schema as well.